tests/unit/components/plate_based_widgets/stimulationstudio/StimulationStudioWidget.spec.js

```diff
@@ -50,117 +47,369 @@ describe("StimulationStudioWidget.vue", () => {
+        shiftKey: true,
+      });
+      array_of_well_indices.map((well) => expect(wrapper.vm.stroke_width[well]).toBe(4));
+    }
+  );
+
+  test("Given that any wells are selected, When user Shift+clicks the Row/Column button or any individual well, Then wells corresponding are seleced with outline of 4px", async () => {
+    const wrapper = mount(StimulationStudioWidget, {
+      store,
+      localVue,
+    });
+    new Array(24).map(async (well) => {
+      await wrapper.find().trigger("click", {
+        shiftKey: true,
+      });
+      expect(wrapper.vm.sroke_width[well]).toBe(4);
+    });
+  });
+
+  test("When there is a change to all_selected wells, Then commit the mutation to state to the store", async () => {
+    const wrapper = mount(StimulationStudioWidget, {
+      store,
+      localVue,
+    });
+    await wrapper.vm.basic_select(3);
+    expect(store.state.stimulation.selected_wells).toStrictEqual([3]);
+  });
+
+  test("When an unselected is hovered over and left, Then it should toggle a stroke with of 2px and 0px", async () => {
+    const wrapper = mount(StimulationStudioWidget, {
+      store,
+      localVue,
+    });
+    const test_wells = new Array(24);
+    test_wells.map(async (well) => {
+      await wrapper.find("#plate_" + well).trigger("enter-well");
+      expect(wrapper.vm.stroke_width[well]).toBe(2);
+      await wrapper.find("#plate_" + well).trigger("leave-well");
+      expect(wrapper.vm.stroke_width[well]).toBe(0);
+    });
+  });
+
+  test("Given that no wells are selected, When user Shift+Click on %s, Then then well %s visually become selected due to %s (have the stroke outline and be mutated to state", async () => {
+    const wrapper = mount(StimulationStudioWidget, {
+      store,
+      localVue,
+    });
+    const check_store = store.state.stimulation.selected_wells.length;
+    new Array(24).map(async (well) => {
+      await wrapper.find("#plate_" + well).trigger("click", {
+        shiftKey: true,
+      });
+      expect(wrapper.vm.sroke_width[well]).toBe(4);
+      expect(check_store).toBe(well);
+    });
   });
+
+  test("Given any number of wells, but all are selected, When plus-minus icon is hovered over and left, Then all unselected wells should have a stroke width of 2", async () => {
+    const wrapper = mount(StimulationStudioWidget, {
+      store,
+      localVue,
+    });
+    await wrapper.find(".span__stimulationstudio-toggle-plus-minus-icon").trigger("mouseenter");
+    wrapper.vm.stroke_width.map((well) => {
+      expect(well).toBe(2);
+    });
+    await wrapper.find(".span__stimulationstudio-toggle-plus-minus-icon").trigger("mouseleave");
+    wrapper.vm.stroke_width.map((well) => {
+      expect(well).toBe(0);
+    });
+  });
+  // test("Given no wells are selected in a row/column, When user clicks and unclicks a row/column label, Then corresponding unselected wells will toggle a stroke-width of 4px and 0px", async () => {
+  //   const propsData = {
```

Comment:
Kept these in case they're useful later, but don't need them until protocol assigned is set up